pysradb/search.py

```diff
@@ -0,0 +1,1619 @@
+            "selection": "Library Selection",
+            "platform": "Platform",
+            "basecount": "Base Count",
+        }
+        if "all" not in graph_types:
+            selected_plots = []
+            for graph_type in graph_types:
+                if graph_type not in plot_keys:
+                    continue
+                for plot in plots:
+                    if plot_keys[graph_type] in plot and plot not in selected_plots:
+                        selected_plots.append(plot)
+            plots = selected_plots
+        too_many_organisms = False
+        if self.stats["graph_raw"]["Organism"].nunique() > 30:
+            print(
+                "Too many types of organisms to plot (>30). Showing only top 30 organisms."
+            )
+            too_many_organisms = True
+        for plot in plots:
+            self._plot_graph(plt, plot, show, saveto, too_many_organisms)
+
+    def search(self):
+        pass
+
+    def get_df(self):
+        """Getter for the search result dataframe."""
+        return self.df
+
+    def _plot_graph(self, plt, axes, show, savedir, too_many_organisms):
+        """Plots a graph based on data from self.stats
+
+        Parameters
+        ----------
+        axes: tuple
+            tuple containing 1 or 2 strings corresponding to the statistics
+            to plot. 1 string: Histogram. 2 string: heat map
+        savedir: str
+            directory to save to
+        show: bool
+            whether to call plt.show
+        """
+
+        if (
+            "Publication Date" in axes
+            and self.stats["graph_raw"]["Publication Date"].nunique() > 30
+        ):
+            self.stats["graph_raw"]["Publication Date"] = self.stats["graph_raw"][
+                "Publication Date"
+            ].str[:-3]
+        if axes == ("Base Count",):
+            count = list(self.stats["count_data"])
+            title = "Histogram of Base Count"
+            plt.figure(figsize=(15, 10))
+            plt.hist(count, min(70, len(count)), color="#135c1c", log=True)
+            plt.xlabel("Base Count", fontsize=14)
+            plt.ylabel("Frequency", fontsize=14)
+            plt.xticks(rotation=90)
+            plt.title(title, fontsize=18)
+            plt.savefig(f"{savedir}{title}.jpeg")
+        elif len(axes) == 1:
+            title = f"Histogram of {axes[0]}"
+            data = self.stats["graph_raw"][axes[0]].value_counts()
+            if too_many_organisms:
+                data = data[:30]
+            plt.figure(figsize=(15, 10))
+            plt.bar(
+                range(len(data.values)),
+                data.values,
+                tick_label=list(data.index),
+                color="#135c1c",
+            )
+            plt.xticks(rotation=90)
+            plt.title(title, fontsize=18)
+            plt.xlabel(axes[0], fontsize=14)
+            plt.ylabel("Frequency", fontsize=14)
+            plt.savefig(f"{savedir}{title}.jpeg")
```

Comment:
I think the figures should be saved invector format by default. Either to a pdf or a svg. The user can create a jpeg/png if required.

Reply:
Resolved in 5104ca6